File: E02/C++/E02.2T03.cpp
/////////////////////////////////////////
// Lab 3.	Restaurant Discount
/////////////////////////////////////////

// Very imprtan information, i don't use Decimal type in the task

#include <iostream>
#include <iomanip>
#include <string>
using namespace std;

int main() {

	string nameHall[] = { "Small Hall", "Terrace", "Great Hall" };
	int capacityHall[] = { 50, 100, 120 };
	float priceHall[] = { 2500, 5000, 7000 };
	string servicePackage[] = { "Normal", "Gold", "Platinum" };
	int discountPackage[] = { 5, 10, 15 };
	float pricePackage[] = { 500, 750, 1000 };

	int count;
	string package;

	cout << "Count people: ";
	cin >> count;
	cout << "Package: ";
	cin.ignore();
	getline(cin, package);

	int indexHall = 0;
	while (count > capacityHall[indexHall]) {
		indexHall++;
	}

	int indexPack = 0;
	for (; indexPack < 3; indexPack++) {						//Hardcoded
		if (!package.compare(servicePackage[indexPack])) {
			break;
		}
	}

	double price = priceHall[indexHall] + pricePackage[indexPack];
	double discount = (double)discountPackage[indexPack] * (price / 100);
	price -= discount;	
	price /= count;

	cout << "We can offer you the " << nameHall[indexHall] << " The price per person is ";
	cout << fixed << setprecision(2) << price << "$" << endl;
	
	system("pause");
	return 0;
}

File: E02/Python/E02.2T03.py
#########################################
# Problem 3. Restaurant Discount
########################################

hallPrice = dict()
hallPrice["Small Hall"] = 2500
hallPrice["Terrace"] = 5000
hallPrice["Great Hall"] = 7500

capacity = dict()
capacity[50] = "Small Hall"
capacity[100] = "Terrace"
capacity[120] = "Great Hall"

discount = dict()
discount["Normal"] = 5
discount["Gold"] = 10
discount["Platinum"] = 15

packPrice = dict()
packPrice["Normal"] = 500
packPrice["Gold"] = 750
packPrice["Platinum"] = 1000

size = input()
pack = input()

keys = list(capacity.keys())

check = False;
for key in keys:
    if key > int(size):
        if discount[pack] != 0 and size != 0:
            total = (int(hallPrice[capacity[key]] + packPrice[pack]) -
                     (int(hallPrice[capacity[key]] + packPrice[pack]) / int(discount[pack]))) / float(size)
        check = True
        print("We can offer you the " + capacity[key] + " The price per person is {0:.2f}$".format(total))
        break

if not check:
    print("We do not have an appropriate hall.")
